test/model/world/world.cpp

```diff
@@ -6,62 +6,86 @@
-            world.addIndividual(new Fred(40, 50, fredE));
-            world.addIndividual(new Fred(40, 50, fredE));
-            world.addIndividual(new Fred(40, 50, fredE));
+            world->addFood(new Food(10, 20, foodE));
+            world->addIndividual(new Fred(40, 50, fredE));
+            world->addIndividual(new Fred(40, 50, fredE));
+            world->addIndividual(new Fred(40, 50, fredE));
 
             THEN ("the object gets stored properly") {
-                REQUIRE(world.getFood().back()->getPosition() == Point(10, 20));
-                REQUIRE(world.getIndividuals().back()->getEnergy() == fredE);
+                REQUIRE(world->getFood().back()->getPosition() == Point(10, 20));
+                REQUIRE(world->getIndividuals().back()->getEnergy() == fredE);
 
-                int ID = 0;
-                for (auto fred : world.getIndividuals()) REQUIRE(fred->getID() == ID++);
+                int ID = world->getIndividuals()[0]->getID();
+                for (auto fred : world->getIndividuals()) REQUIRE(fred->getID() == ID++);
             }
 
             THEN ("the amount of things in the world increases") {
-                REQUIRE(world.getFood().size() == 1);
-                REQUIRE(world.getIndividuals().size() == 3);
+                REQUIRE(world->getFood().size() == 1);
+                REQUIRE(world->getIndividuals().size() == 3);
             }
 
             THEN ("the energy level rises") {
-                REQUIRE(world.getEnergy() == foodE + 3 * fredE);
+                REQUIRE(world->getEnergy() == foodE + 3 * fredE);
             }
 
             WHEN ("I fill up to max ENERGY") {
-                world.fillWithFood([](const World * world, int energy) -> Food * {
+                world->fillWithFood([](const World * world, int energy) -> Food * {
                     return new Food(0, 0, std::min(foodE, energy));
                 });
 
                 THEN ("the amount of food and energy rises further") {
-                    REQUIRE(world.getIndividuals().size() == 3);
+                    REQUIRE(world->getIndividuals().size() == 3);
 
-                    REQUIRE(world.getFood().size() > 1);
-                    REQUIRE(world.getEnergy() == world.ENERGY);
+                    REQUIRE(world->getFood().size() > 1);
+                    REQUIRE(world->getEnergy() == world->ENERGY);
                 }
 
                 THEN ("adding more entities raises an overflow exception") {
-                    REQUIRE_THROWS_AS(world.addFood(new Food(0, 0, 10)), std::overflow_error);
-                    REQUIRE_THROWS_AS(world.addIndividual(new Fred(0, 0, 10)), std::overflow_error);
+                    REQUIRE_THROWS_AS(world->addFood(new Food(0, 0, 10)), std::overflow_error);
+                    REQUIRE_THROWS_AS(world->addIndividual(new Fred(0, 0, 10)), std::overflow_error);
                 }
             }
         }
     }
-}
\ No newline at end of file
+}
+
+SCENARIO("The world can freely be modified") {
```

Comment:
I'm not quite sure that this should be a separate scenario. If you share variables between scenarios, it should probably be only one.

Reply:
true